### PR TITLE
Added new Service to update trade pairs from exchange

### PR DESCRIPTION
**Pull request by mestorlx**
This PR addresses issue #87
The new service `pairDiscoveryService` will periodically fetch trade pairs available from all exchanges.
Each exchange scraper needs to implement `FetchAvailablePairs() []string`.
Each scrapper should implement the method to retrieve the paring list from exchange. The service will call these methods.
Regarding the storage in redis I believe there is no current structure for holding the available pairs. I was thinking in making lists where the key is the exchange name (string) and each element is a standardized string with the pair such as “BTC_USD"

### CODE REVIEW

File: cmd/services/pairDiscoveryService/main.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		if err == nil {
+			s = scrapers.NewAPIScraper(e, c.ApiKey, c.SecretKey)
+		} else {
+			log.Error("Error processing config for exchange:"+e+" error:", err.Error())
+			s = scrapers.NewAPIScraper(e, "", "")
+		}
+		if s != nil {
+			p, err := s.FetchAvailablePairs()
+			if err == nil {
+				err := db.AddAvailablePairsExchange(e, p)
```
Comment (frranck) [inline cmd/services/pairDiscoveryService/main.go]: would call it Set to be consistent

File: internal/pkg/exchange-scrapers/CoinBaseScraper.go [hunk 2]
```diff
@@ -134,6 +137,31 @@ func (s *CoinBaseScraper) Close() error {
...
+	var ar []gdax.Product
+	err = json.Unmarshal(data, &ar)
+	if err == nil {
+		pairs = make([]dia.Pair, len(ar))
+		log.Print("Retrieved:", len(ar))
+		for i, p := range ar {
+			pairs[i] = dia.Pair{
+				Symbol:      p.BaseCurrency,
+				ForeignName: p.Id,
+				Exchange:    dia.CoinBaseExchange,
```
Comment (frranck) [inline internal/pkg/exchange-scrapers/CoinBaseScraper.go]: maybe would be good occasion to get rid of this constant, like by passing the info in the NewScrapper ?

File: pkg/dia/helpers/NameForSymbol.go [hunk 3]
```diff
@@ -359,7 +457,7 @@ func NameForSymbol(symbol string) string {
 	case "SYS":
 		return "Syscoin"
 	case "BTM":
-		return "Bytom"
+		return "Bitmark"
 	case "HAL":
```
Comment (frranck) [inline pkg/dia/helpers/NameForSymbol.go]: this one has is actually Bytom https://coinhub.diadata.org/coin-details/BTM

**Comment by frranck** [comment]
Amazing work thanks,
I'm a bit worried about the changes into NameForSymbol thought.

File: pkg/dia/helpers/NameForSymbol.go (hunk 3 above)
Comment (mestorlx) [reply pkg/dia/helpers/NameForSymbol.go]: Actually is [duplicated](https://coinmarketcap.com/search/?q=BTM).
If you want I can update.

**Comment by frranck** [comment]
Also could you add a wait so it updates the pairs only once every hour ? some exchanges have quotas...

**Pull request merged**